### PR TITLE
Humanization

### LINKED ISSUE #7606: We should make "friendly field names" way less aggressive
"Friendly field names" has been a persistent source of confusion and annoyance for our users. In too many cases, it guesses wrong and produces results that look very broken, even when table/field names are in English.

I think we should refactor the behavior of this setting to only capitalize the first letter of each word in table/field names, and replace underscores with spaces. If that's all it did, I could see it being okay for it to be on by default.

@metabase/core-developers — any different opinions?

### LINKED ISSUE #9908: Humanization of column names is way too aggressive
On sql queries, things get kind crazy 
Screen Shot 2019-05-07 at 1 24 36 PM

### PR DESCRIPTION
**Pull request by ogeagla**
Resolves #9908
Resolves #7606

 - [x] Fix user cases 
 - [x] Remove junky/not-so-friendly words 
 - [x] Reverse word dict to favor rarer words first
 - [x] Remove small word fragments
 - [x] Move articles / pronouns / conjunctions to end (I, the, an, etc) to make them last things to try
 - [x] Test Metabase fields on new algo
 - [x] Inspect cost graph for corpus and see if there's obvious inflection point

### CODE REVIEW

**Comment by tlrobinson** [comment]
We might want to cross-reference that word list with an actual English dictionary and remove any other obviously dumb words (I generated that frequency list from a dump of the English Wikipedia without doing much checking).

Also, `to` is a word that could plausibly appear in a field name. Maybe we should keep it but move it further down the list.

(Longer term, I really wish we had a larger set of data of people manually renaming fields in the "data model" admin settings that we could use to test or train this better)

**Comment by sbelak** [comment]
Low hanging fruit is to first tokenize on underscores and capital letters and and take out sequences of capital letters (almost certainly an acronym).

**Comment by sbelak** [comment]
I'd look at the distribution (probably cum-sum) of cost for good vs. bad splits, probably nomralized by length in some form (input length seems the most sane, but I wouldn't be surprised if it ends up being square of length or number of output tokens or something like that). My hunch is that the difference between good and bad is pretty stark and we could have a good static cutoff threshold after which we don't bother.

This is in the realm of low-hanging fruit, next step would be language detection, but if we're only doing English or nothing, the above should suffice.

**Review by camsaul — APPROVED** [review]
nice. LGTM

**Comment by tlrobinson** [comment]
I'm wondering if it would be safe to automatically uppercase ~2-3 letter words that don't appear in the wordlist on the assumption that they are acronyms?

**Comment by camsaul** [comment]
I would vote against trying to make it more magical and doing more things rn

File: test/metabase/models/humanization_test.clj [hunk 1]
```diff
@@ -34,6 +34,57 @@
...
+(expect "Changelog"                   (humanization/name->human-readable-name :advanced "changelog"))
+(expect "Dataflow"                    (humanization/name->human-readable-name :advanced "dataflow"))
+(expect "Bestseller"                  (humanization/name->human-readable-name :advanced "bestseller"))
+(expect "Uniques"                     (humanization/name->human-readable-name :advanced "uniques"))
+(expect "Cpi"                         (humanization/name->human-readable-name :advanced "cpi"))
+(expect "Ctr"                         (humanization/name->human-readable-name :advanced "ctr"))
+(expect "Paid To Sparkify"            (humanization/name->human-readable-name :advanced "paid_to_sparkify"))
+(expect "Paid To Sparkify"            (humanization/name->human-readable-name :advanced "paidtosparkify"))
+(expect "Assassinate"                 (humanization/name->human-readable-name :advanced "assassinate"))
+(expect "An Album Cover"              (humanization/name->human-readable-name :advanced "analbumcover"))
```
Comment (dataxpress) [inline test/metabase/models/humanization_test.clj]: :+1:

**Pull request merged**